specification/search/data-plane/Microsoft.Azure.Search.Data/readme.md

```diff
@@ -101,3 +101,23 @@ csharp:
   clear-output-folder: true
   output-folder: $(csharp-sdks-folder)/Search/DataPlane/Microsoft.Azure.Search.Data/Generated
 ```
+
+### Tweak generated code
+
+Use the below directives sparingly. Every modification we make in here, we potentially have to replicate for every target language.
+
+``` yaml $(csharp)
+directive:
+  # Make all Proxy types internal so we can version them freely.
+  - from: source-file-csharp
+    where: $
+    transform: return $.replace( /public partial interface IDocumentsProxyOperations/g, "internal partial interface IDocumentsProxyOperations" )
+
+  # Change the public property on ISearchIndexClient and SearchIndexClient to refer to our public interface instead of the internal one.
```

Comment:
Is it worth adding a note, *why* in addition to the *what* ?

Reply:
There is a broader context behind the _why_, and I'm not sure this is the best place to explain it, but it won't hurt to add a bit more detail.

Reply:
After taking a second look, I realized this transform could be simplified, so I did that too. I'll have a corresponding SDK PR once this has been merged (codegen will fail without it).

Reply:
Makes sense, thanks!